chore: Add setRequiredFields and setIgnoredFields documentation (#911)
SAP/cloud-sdk: docs-js/features/odata/common/operations/set-required-fields.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> If updating an entity requires a specific field, even though it didn't change, you can use the `setRequiredFields` method.

Reviewer: ```suggestion
If you want to send properties in the payload of the update request, that you did not change, use the `setRequiredFields` method to add them.
```

> In the following example, the `PATCH` request will contain the `FIRST_NAME` property of the business partner, even if it didn't change.

Reviewer: [nth] Same suggestion as for ignore.

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

If you want to send properties in the payload of the update request, that you did not change, use the `setRequiredFields` method to add them.

In the following example, the `PATCH` request will contain the `FIRST_NAME` property of the business partner, even if it didn't change.

<Tabs groupId="version" defaultValue="major" values={[ { label: 'SDK 2.x', value: 'major' }, { label: 'SDK 1.x', value: 'deprecated' } ]}>
<TabItem value="major">

```ts
const { businessPartnerApi } = businessPartnerService();

> // First we get a businesspartner with the id of "1".

Reviewer: ```suggestion
// Get a business partner
```

const businessPartner = businessPartnerApi
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

> // Then we change the business partner's last name to "Smith".

Reviewer: ```suggestion
// Change last name
```

businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.lastName = 'Smith'));

> // Finally we update the business partner with the updated last name, but still include their first name in the "PATCH" request.

Reviewer: What do you think?

```suggestion
// Send a PATCH request with `{ "LastName" : "Smith", "FirstName": "Steve" }`  and include the first name although it was not changed
```

businessPartnerApi
  .requestBuilder()
  .update(businessPartner)
  .setRequiredFields(businessPartnerApi.schema.FIRST_NAME);
```

</TabItem>
<TabItem value="deprecated">

```ts

> // First we get a businesspartner with the id of "1".

Reviewer: ```suggestion
// Get a business partner
```

const businessPartner = BusinessPartner
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

> // Then we change the business partner's last name to "Smith".

Reviewer: ```suggestion
// Change last name
```

businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.lastName = 'Smith'));

> // Finally we update the business partner with the updated last name, but still include their first name in the "PATCH" request.

Reviewer: What do you think?

```suggestion
// Send a PATCH request with `{ "LastName" : "Smith", "FirstName": "Steve" }`  and include the first name although it was not changed
```
